feat: Add btn to finish adding new product

WildOrangutan opened the pull request:
### What
Add btn on the end of product page, when adding a new product.

Marking as draft for now - waiting #4698 to merge and resolve conflicts after.

### Screenshot
![image](https://github.com/openfoodfacts/smooth-app/assets/5972966/1428ef16-58e7-44c0-9174-40be1dd8ca2a)

### Fixes bug(s)
Fixes #4670

WildOrangutan committed:
feat: Add btn to finish adding new product

M123-dev commented:
Heyy @WildOrangutan the screenshot looks good to me, why is this Pull Request wip

WildOrangutan commented:
@M123-dev it's ready. I marked it as a draft, because [you said to wait for the other MR](https://github.com/openfoodfacts/smooth-app/issues/4670#issuecomment-1750765540) 🙂

I will mark it as ready.

M123-dev commented on packages/smooth_app/lib/pages/product/add_new_product_page.dart:
```diff
@@ -469,4 +479,67 @@ class _AddNewProductPageState extends State<AddNewProductPage>
+                    padding: const EdgeInsets.all(SMALL_SPACE),
+                    child: Text(
+                      localizations.new_product_done_msg(
+                          OpenFoodAPIConfiguration.globalUser?.userId ??
+                              localizations.user_profile_title_guest),
```
"user_profile_title_guest" will just be `Welcome!`

The sentence 

Thanks for your contribution "welcome!"! doesn't make sense.
I'd suggest to create a new no user string in  app_en.arb

Something like

`"new_product_done_msg_no_user": "Thanks for your contribution !",`

M123-dev commented on packages/smooth_app/lib/pages/product/add_new_product_page.dart:
```diff
@@ -469,4 +479,67 @@ class _AddNewProductPageState extends State<AddNewProductPage>
+                                ),
+                                const SizedBox(width: SMALL_SPACE),
+                                const Icon(Icons.arrow_forward),
+                              ])),
+                    )),
```
Please add some commas here between the braces

M123-dev commented on packages/smooth_app/lib/pages/product/add_new_product_page.dart:
```diff
@@ -469,4 +479,67 @@ class _AddNewProductPageState extends State<AddNewProductPage>
+                    )
+                  ],
+                )
+              ])),
```
Also some commas here, the analyzer doesn't complain but that will make the code clearer to read

M123-dev reviewed (commented):
Heyy @WildOrangutan sorry for the late review, I added a small comment regarding the translation string, otherwise the code looks good to me. Feel free to ping me when its fixed then we can merge before the October ends

WildOrangutan commented on packages/smooth_app/lib/pages/product/add_new_product_page.dart:
```diff
@@ -469,4 +479,89 @@ class _AddNewProductPageState extends State<AddNewProductPage>
+  }
+}
+
+String? _getRegisteredUserName() {
+  final String? userId = OpenFoodAPIConfiguration.globalUser?.userId;
+  if (userId == null || userId.toLowerCase().contains('welcome')) {
+    return null;
+  } else {
+    return userId;
+  }
 }
```
Any more appropriate place to put this helper function?

M123-dev replied:
Don't think so, we have a similar code for the settings page but its fine to leave it here

WildOrangutan replied:
Ok will leave it here for now. I guess it could be moved to `UserManagementHelper`, so we can remove this duplicate logic.

WildOrangutan commented:
> Heyy @WildOrangutan sorry for the late review, I added a small comment regarding the translation string, otherwise the code looks good to me. Feel free to ping me when its fixed then we can merge before the October ends

@M123-dev no worries. I've updated with suggested changes. Please check again.

M123-dev commented on packages/smooth_app/lib/pages/product/add_new_product_page.dart:
```diff
@@ -469,4 +479,89 @@ class _AddNewProductPageState extends State<AddNewProductPage>
+
+String? _getRegisteredUserName() {
+  final String? userId = OpenFoodAPIConfiguration.globalUser?.userId;
+  if (userId == null || userId.toLowerCase().contains('welcome')) {
+    return null;
+  } else {
+    return userId;
```
When does the username contain welcome, and why should we not allow it

WildOrangutan replied:
Since guest user will have name "Welcome!"? You have any other suggestions, how to take this into account?

WildOrangutan replied:
I think I misunderstood your previous [comment](https://github.com/openfoodfacts/smooth-app/pull/4706#discussion_r1376495964). Will do same name-check logic as on settings page.

M123-dev replied:
Ohhh yes, seeing at from that point my previous comment was a bit confusing

M123-dev commented on packages/smooth_app/lib/pages/product/add_new_product_page.dart:
```diff
@@ -469,4 +479,89 @@ class _AddNewProductPageState extends State<AddNewProductPage>
+
+String? _getRegisteredUserName() {
+  final String? userId = OpenFoodAPIConfiguration.globalUser?.userId;
+  if (userId == null || userId.toLowerCase().contains('welcome')) {
+    return null;
+  } else {
+    return userId;
```
Also it would be good to catch the case if `userId.isEmail``and show the default "no user" string

M123-dev reviewed (approved):
Perfect, thanks a lot @WildOrangutan
